test/rateLimiters/slidingWindowCounter.test.ts

```diff
@@ -0,0 +1,476 @@
+
+            const initRequest = 8;
+
+            // large request at very end of first fixed window
+            await limiter.processRequest(user4, timestamp + WINDOW_SIZE, initRequest);
+
+            // 9 + 8 * .25 = 11, over capacity (request should be blocked)
+            expect(
+                (await limiter.processRequest(user4, timestamp + WINDOW_SIZE * 1.75, 9)).tokens
+            ).toBe(10);
+
+            // currentTokens (in current fixed window): 0
+            // previousTokens (in previous fixed window): 8
+            const count = await getWindowFromClient(client, user4);
+            expect(count.currentTokens).toBe(4);
+            expect(count.previousTokens).toBe(initRequest);
+        });
+    });
+
+    describe('SlidingWindowCounter functions as expected', () => {
+        afterEach(() => {
+            client.flushall();
+        });
+
+        test('allows user to consume current allotment of tokens', async () => {
+            // "free requests"
+            expect((await limiter.processRequest(user1, timestamp, 0)).success).toBe(true);
+            // Test 1 token requested
+            expect((await limiter.processRequest(user1, timestamp, 1)).success).toBe(true);
+            // Test < CAPACITY tokens requested
+            expect((await limiter.processRequest(user2, timestamp, CAPACITY - 1)).success).toBe(
+                true
+            );
+            // <= CAPACITY tokens requested
+            expect((await limiter.processRequest(user3, timestamp, CAPACITY)).success).toBe(true);
+        });
+
+        test('blocks exceeding requests over token allotment', async () => {
+            // Test > capacity tokens requested
+            expect((await limiter.processRequest(user1, timestamp, CAPACITY + 1)).success).toBe(
+                false
+            );
+
+            // Fill up user 1's window
+            const value: RedisWindow = {
+                currentTokens: 10,
+                previousTokens: null,
+                fixedWindowStart: timestamp,
+            };
+            await client.set(user1, JSON.stringify(value));
+
+            // window is full. Shouldn't be allowed to take 1 token
+            expect((await limiter.processRequest(user1, timestamp, 1)).success).toBe(false);
+
+            // Should still be allowed to process "free" requests
+            expect((await limiter.processRequest(user1, timestamp, 0)).success).toBe(true);
+        });
+
+        test('fixed window and current/previous tokens update as expected', async () => {
+            await setTokenCountInClient(client, user1, 0, null, timestamp);
+            // fills first window with 4 tokens
+            await limiter.processRequest(user1, timestamp, 5);
+            // fills second window with 5 tokens
+            expect(
+                await (
+                    await limiter.processRequest(user1, timestamp + WINDOW_SIZE + 1, 4)
+                ).tokens
+            ).toBe(6);
```

Comment:
Is the response providing the number of tokens available in the rolling window? If so then I would expect the result to be `1` instead of `6` which is the number of tokens available in the current window depending on how you're rounding percentages? 

Are you rounding down or up?

Reply:
I see was you updated to `2` in the implementation PR. Seems like you're rounding down. No change needed in that case.